torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -729,6 +734,29 @@ void Reducer::finalize_backward() {
   local_used_maps_reduced_ = false;
 }
 
+void runGradCallbaclForVariable(
+    torch::autograd::Variable& variable,
+    GradCallback&& cb) {
+  auto context_ptr = rpc_context_.context_ptr.load();
+  if (context_ptr == nullptr) {
+    cb(variable.grad());
+  } else {
+    // Under distributed autograa
+    context_ptr->runGradCallbaclForVariable(std::move(cb));
+  }
+}
+
+void Reducer::RpcContext::set(ContextPtr::weak_type context_weak_ptr) {
+  auto new_context_ptr context_weak_ptr.lock();
+  TORCH_INTERNAL_ASSERT(new_context_ptr);
```

Comment:
This assert will fail if we're running with local autograd?

Reply:
> This assert will fail if we're running with local autograd?

I'll keep that in mind.

Reply:
Fixed